proto/desmos/posts/v1beta1/msgs.proto

```diff
@@ -0,0 +1,198 @@
+syntax = "proto3";
+package desmos.posts.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "desmos/posts/v1beta1/posts.proto";
+import "desmos/posts/v1beta1/polls.proto";
+
+option go_package = "github.com/desmos-labs/desmos/x/posts/types";
+
+// Msg defines the relationships Msg service.
+service Msg {
+  // CreatePost defines the method to create a post
+  rpc CreatePost(MsgCreatePost) returns (MsgCreatePostResponse);
+
+  // EditPost defines the method to edit an existing post
+  rpc EditPost(MsgEditPost) returns (MsgEditPostResponse);
+
+  // AddReaction defines the method to add a reaction to a post
+  rpc AddPostReaction(MsgAddPostReaction) returns (MsgAddPostReactionResponse);
+
+  // RemoveReaction defines the method to remove a reaction from a post
+  rpc RemovePostReaction(MsgRemovePostReaction) returns (MsgRemovePostReactionResponse);
+
+  // RegisterReaction defines the method to register a new reaction
+  rpc RegisterReaction(MsgRegisterReaction) returns (MsgRegisterReactionResponse);
+
+  // AnswerPoll defines the method to answer a poll
+  rpc AnswerPoll(MsgAnswerPoll) returns (MsgAnswerPollResponse);
+}
+
+// ___________________________________________________________________________________________________________________
+
+// MsgCreatePost represents the message to be used to create a post.
+message MsgCreatePost {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string parent_id = 1 [
+    (gogoproto.customname) = "ParentID",
+    (gogoproto.moretags) = "yaml:\"parent_id\""
+  ];
+  string message = 2 [
+    (gogoproto.moretags) = "yaml:\"message\""
+  ];
+  bool allows_comments = 3 [
+    (gogoproto.jsontag) = "allows_comments",
+    (gogoproto.moretags) = "yaml:\"allows_comments\""
+  ];
+  string subspace = 4 [
+    (gogoproto.moretags) = "yaml:\"subspace\""
+  ];
+  repeated desmos.posts.v1beta1.OptionalDataEntry optional_data = 5 [
+    (gogoproto.castrepeated) = "OptionalData",
+    (gogoproto.nullable) = false,
```

Comment:
No, in this case the `(gogoproto.nullable)` flag refers to the type contained inside the slice. If we set it to `true`, the result would be a `[]*OptionalDataEntry` instead of `[]OptionalData`. However, this would have cause a cast exception since `OptionalData` is defined as `[]OptionalDataEntry` and not `[]*OptionalDataEntry`. 

Also, by default, all slices are `nullable` (which means that can be set as empty).